src/librustdoc/html/render/context.rs

```diff
@@ -289,7 +289,7 @@ impl<'tcx> Context<'tcx> {
 
         // We can safely ignore synthetic `SourceFile`s.
         let file = match item.span.filename(self.sess()) {
-            FileName::Real(ref path) => path.local_path().to_path_buf(),
+            FileName::Real(ref path) => path.local_path()?.to_path_buf(),
```

Comment:
Ditto, this looks wrong. Rustdoc still needs to generate links to non-local files.